service/src/main/java/uk/gov/hmcts/reform/fpl/validators/HasChildrenNameValidator.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.fpl.validators;
+
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.Children;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasChildName;
+
+import java.util.Objects;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class HasChildrenNameValidator implements ConstraintValidator<HasChildName, Children> {
+
+    @Override
+    public void initialize(HasChildName constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(Children children, ConstraintValidatorContext constraintValidatorContext) {
+
+        if (children.getAdditionalChildren() != null || children.getFirstChild() != null) {
+            return children.getAllChildren().stream()
```

Comment:
I don't like the fact the other fields are checked in if statement and other method is used to build stream, I know what you wanted to achieve, but now this validator checks two things:

1. If there is any Child, return false if not
2. If there is, check if all children have got name. 

Isn't the first another validator responsibility?

Reply:
Yes you're right but the validation message remains the same and we would have to do a null chuck in 2 anyway. Is it not better to do both in the one?

Reply:
@akomisarek

Reply:
Sure, just an idea. It's fine to leave it!